File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/board/domain/bookmark/service/WriteBookmarkService.java
package io.dustin.apps.board.domain.bookmark.service;

import io.dustin.apps.board.domain.bookmark.model.Bookmark;
import io.dustin.apps.board.domain.bookmark.repository.BookmarkRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Service
public class WriteBookmarkService {

    private final BookmarkRepository bookmarkRepository;

    public Bookmark create(Long userId, Long boardId){
        Bookmark bookmark = Bookmark.builder()
                .userId(userId)
                .boardId(boardId)
                .build();
        this.bookmarkRepository.save(bookmark);
        return bookmark;
    }

    @Transactional
    public void delete(Long userId, Long boardId){
        this.bookmarkRepository.deleteByUserIdAndBoardId(userId, boardId);
    }


}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/bookmark/service/WriteBookmarkService.kt
package io.dustin.apps.board.domain.bookmark.service

import io.dustin.apps.board.domain.bookmark.model.Bookmark
import io.dustin.apps.board.domain.bookmark.repository.BookmarkRepository
import org.springframework.stereotype.Service

@Service
class WriteBookmarkService (
    private val bookmarkRepository: BookmarkRepository
) {

    /**
     * 북마크 추가
     */
    fun create(userId: Long, boardId: Long): Bookmark {
        val bookmark = Bookmark(
            userId = userId,
            boardId = boardId
        )
        bookmarkRepository.save(bookmark)
        return bookmark
    }

    /**
     * 북마크 해제
     */
    fun delete(userId: Long, boardId: Long) {
        bookmarkRepository.deleteByUserIdAndBoardId(userId, boardId)
    }
}
